frontend/src/components/Pages/InspectionPage/InspectionSection.tsx

```diff
@@ -114,82 +170,137 @@ export function InspectionSection({ refreshInterval }: RefreshProps) {
         setSelectedMissions(inspections.map((i) => i.missionDefinition))
     }
 
-    const updateOngoingMissionsMap = async (areaMissions: DeckMissionType) => {
-        let newOngoingMissions: OngoingMissionType = {}
-        for (const areaId of Object.keys(areaMissions)) {
-            for (const inspection of areaMissions[areaId].inspections) {
-                const missionDefinition = inspection.missionDefinition
-                const missionRuns = await BackendAPICaller.getMissionRuns({
-                    statuses: [MissionStatus.Paused, MissionStatus.Pending, MissionStatus.Ongoing],
-                    missionId: missionDefinition.id,
-                })
-                newOngoingMissions[missionDefinition.id] = missionRuns.content.length > 0
-            }
-        }
-        setOngoingMissions(newOngoingMissions)
-    }
-
     useEffect(() => {
         const id = setInterval(() => {
             updateOngoingMissionsMap(deckMissions)
         }, refreshInterval)
         return () => clearInterval(id)
     }, [deckMissions])
 
+    const getDeadlineInspection = (deadline: Date) => {
+        const deadlineDays = getDeadlineInDays(deadline)
+        switch (true) {
```

Comment:
If the switch parameter is "true" and not a parameter I suggest just using if sentences instead

Reply:
I think this is the reason why Anders used the switch before:

"A switch statement is significantly faster than an if-else ladder if there are many nested if-else's involved. This is due to the creation of a jump table for switch during compilation. As a result, instead of checking which case is satisfied throughout execution, it just decides which case must be completed. The number of comparisons made is lesser hence, reducing the compile time. "

Reply:
btw there was a mistake here, that i have corrected:
![image](https://github.com/equinor/flotilla/assets/108514046/6fa0802f-ed5a-4e30-9a91-8682ce2c6717)

Reply:
> I think this is the reason why Anders used the switch before:
> 
> "A switch statement is significantly faster than an if-else ladder if there are many nested if-else's involved. This is due to the creation of a jump table for switch during compilation. As a result, instead of checking which case is satisfied throughout execution, it just decides which case must be completed. The number of comparisons made is lesser hence, reducing the compile time. "

It's also more concise and easier to read imo. It's not unusual to do in typescript either